### PR TITLE
Add skipRemoveUsers and skipDeleteChannel

### PR DESCRIPTION
**Pull request by artificial-aidan**
Would like to be able to skip user management and channel deletion for certain use cases. Thought it would be useful to upstream, have been testing it locally.

### CODE REVIEW

**Comment by stakater-user** [comment]
@artificial-aidan Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/a7920a384430093c01b13fb5bd61a221a36236b5/checks) has Failed!

**Comment by rasheedamir** [comment]
@artificial-aidan can you plz check the failed pipeline?

**Comment by artificial-aidan** [comment]
When I checked yesterday it was a 137 exit code for golint. Which is usually OOM killed.

**Comment by artificial-aidan** [comment]
Confirmed that lint passes locally. It appears to have been killed by an OOM condition on the build.

**Comment by stakater-user** [comment]
@artificial-aidan Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/d532ce929189bcafa23bf756ffe9eb97a101d4f7/checks) has Failed!

**Comment by stakater-user** [comment]
@artificial-aidan Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/9a64c044a2ad384d44c033e2d57cb7d0df970846/checks) has Failed!

**Comment by stakater-user** [comment]
@artificial-aidan Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/b03b5edbc8b2998d338b8b4932fb6090a43dc3c1/checks) has Failed!

**Comment by stakater-user** [comment]
@artificial-aidan Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/056bc414a8a40b834fe9a0f5192f03ad501cb6d4/checks) has Failed!

**Comment by artificial-aidan** [comment]
@rasheedamir I upgraded the linter and this made it [pass](https://github.com/artificialinc/slack-operator/actions/runs/3236021348/jobs/5301247142). It will not pass here because the pull request action is based off the base branch.

I upgraded it because of [this comment](https://github.com/golangci/golangci-lint-action/issues/552#issuecomment-1228315360)

**Comment by stakater-user** [comment]
@artificial-aidan Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/090b614de6e6b9e92cead11c9165bd93b9cd38f1/checks) has Failed!

File: api/v1alpha1/channel_types.go [hunk 1]
```diff
@@ -42,6 +42,14 @@ type ChannelSpec struct {
...
 	// +optional
 	Topic string `json:"topic,omitempty"`
+
+	// Don't remove users if they aren't listed.
+	// +optional
+	SkipRemoveUsers bool `json:"skipRemoveUsers,omitempty"`
+
+	// Don't delete the channel when the object is deleted
+	// +optional
+	SkipDeleteChannel bool `json:"skipDeleteChannel,omitempty"`
```
Comment (MuneebAijaz) [inline api/v1alpha1/channel_types.go:52]:
```suggestion
	SkipArchiveChannel bool `json:"skipArchiveChannel,omitempty"`
```
can it be changed to `archive` to be consistent with the terminologies? Rest everything looks fine.

Comment (artificial-aidan) [reply api/v1alpha1/channel_types.go:52]: Shoot. I have this deployed all over the place already. If I get some time I will change it, but unlikely :(

**Review by tanalam2411 — CHANGES_REQUESTED** [review]
Need few feedback to be handled

**Comment by stakater-user** [comment]
@artificial-aidan Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/79d54305da05cea7befbbd028a0a15f1441647b1/checks) has Failed!

**Pull request closed without merging**